tests/e2e/scripts/gke/deploy.sh

```diff
@@ -114,8 +120,21 @@ NAMESPACE="${UNIQUE_ID}"
 run kubectl create namespace "${NAMESPACE}" || error_exit "Namespace already exists"
 
 if [ "${BACKEND}" == 'bookstore' ]; then
+  # Service account key secret.
   get_test_client_key "e2e-non-gcp-instance-proxy-rt-sa.json" "${SA_CRED_PATH}"
   run kubectl create secret generic service-account-cred --from-file="${SA_CRED_PATH}" --namespace "${NAMESPACE}"
+
+  # Generate untrusted self-signed cert.
+  # Common name doesn't matter, client will not verify it.
+  openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
+      -keyout ./server.key -out ./server.crt \
+      -subj "/C=US/ST=CA/O=Google/CN=fake-fqdn.cloud.google.com"
```

Comment:
Are you sure about this CN?   How do we call the gke service?

Reply:
From my testing, `curl -k` will not even verify the CN. Let's see if this passes e2e tests, I think it should work.

Reply:
Works as expected.